<!--
  Copyright 2016 The Cartographer Authors

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->

<launch>
    <param name="/use_sim_time" value="true" />
    <arg name="use_bag" default="true" />

    <param name="robot_description"
        textfile="$(find uav_localization)/urdf/uav_super.urdf" />

    <node name="robot_state_publisher" pkg="robot_state_publisher"
        type="robot_state_publisher" />

    <node name="cartographer_node" pkg="cartographer_ros"
        type="cartographer_node" args="
            -configuration_directory $(find uav_localization)/config
            -configuration_basename mid360_3d.lua"
        output="screen">
        <remap from="points2" to="/livox/lidar_cloud"/>
        <remap from="imu" to="/livox/imu"/>
    </node>

    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
        type="cartographer_occupancy_grid_node" args="-resolution 0.5" />
    
    <node pkg="uav_localization" type="livox_to_pointcloud2.py" name="livox_to_pointcloud2" output="screen"/>

    <group if="$(arg use_bag)">
	    <node name="playbag" pkg="rosbag" type="play" args="--clock /mnt/nvme/rosbag/float2.bag" />
	</group>

    <node name="rviz" pkg="rviz" type="rviz" required="true"
        args="-d $(find uav_localization)/config/mapping_3d.rviz" />
</launch>
